### PR TITLE
Search UAA when adding member to a project.

### PR DESCRIPTION
**Pull request by reasonerjt**
1)Enable UAA client to search UAA by calling '/Users' API.
2)Implement 'SearchUser' in UAA auth helper, register it to auth
package.

### CODE REVIEW

File: src/common/utils/uaa/client.go [hunk 1]
```diff
@@ -56,21 +60,43 @@ type UserInfo struct {
...
 
 func (dc *defaultClient) PasswordAuth(username, password string) (*oauth2.Token, error) {
-	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, dc.httpClient)
-	return dc.oauth2Cfg.PasswordCredentialsToken(ctx, username, password)
+	return dc.oauth2Cfg.PasswordCredentialsToken(dc.prepareCtx(), username, password)
 }
 
 func (dc *defaultClient) GetUserInfo(token string) (*UserInfo, error) {
-	userInfoURL := dc.endpoint + "/uaa/userinfo"
+	userInfoURL := dc.endpoint + "/userinfo"
```
Comment (stonezdj) [inline src/common/utils/uaa/client.go]: Suggest extracting constant

File: src/common/utils/uaa/client.go [hunk 2]
```diff
@@ -92,6 +118,45 @@ func (dc *defaultClient) GetUserInfo(token string) (*UserInfo, error) {
...
+	resp, err := dc.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode > 299 {
```
Comment (stonezdj) [inline src/common/utils/uaa/client.go]: Make sure statusCode<=299 doesn't contain error

File: src/common/utils/uaa/client.go [hunk 3]
```diff
@@ -125,14 +190,21 @@ func NewDefaultClient(cfg *ClientConfig) (Client, error) {
 		ClientID:     cfg.ClientID,
 		ClientSecret: cfg.ClientSecret,
 		Endpoint: oauth2.Endpoint{
-			TokenURL: url + "/uaa/oauth/token",
-			AuthURL:  url + "/uaa/oauth/authorize",
+			TokenURL: url + "/oauth/token",
```
Comment (stonezdj) [inline src/common/utils/uaa/client.go]: Extract constant

File: src/ui/auth/uaa/uaa.go [hunk 4]
```diff
@@ -43,51 +38,85 @@ func GetClient() (uaa.Client, error) {
...
+		user := &models.User{
+			Username: m.Principal,
 			Password: "1234567ab",
-			Email:    username + "@placeholder.com",
-			Realname: username,
-		}
-		err = dao.OnBoardUser(u)
-		if err == nil {
-			return u, nil
+			Email:    m.Principal + "@uaa.placeholder",
```
Comment (stonezdj) [inline src/ui/auth/uaa/uaa.go]: 1) Is it possible to take email address  "test@example.com" as  UAA's principal if it is true, do not create email address
2) Is it possible that username or email contains any leading or trailing space?

File: src/ui/auth/uaa/uaa.go [hunk 5]
```diff
@@ -43,51 +38,85 @@ func GetClient() (uaa.Client, error) {
...
-// 	panic("not implemented")
-// }
+	if len(l) == 0 {
+		return nil, nil
+	}
+	if len(l) > 1 {
+		return nil, fmt.Errorf("Multiple entries found for username: %s", username)
+	}
+	e := l[0]
+	email := username + "@uaa.placeholder"
```
Comment (stonezdj) [inline src/ui/auth/uaa/uaa.go]: Suggest to move them to OnBoardUser() to avoid unformatted data

File: src/ui/auth/uaa/uaa.go (hunk 4 above)
Comment (reasonerjt) [reply src/ui/auth/uaa/uaa.go]: I'll open a bug to better handle the Email.

File: src/ui/auth/uaa/uaa.go (hunk 5 above)
Comment (reasonerjt) [reply src/ui/auth/uaa/uaa.go]: will do , thanks!

File: src/common/utils/uaa/client.go (hunk 1 above)
Comment (reasonerjt) [reply src/common/utils/uaa/client.go]: will do.

File: src/common/utils/uaa/client.go (hunk 3 above)
Comment (reasonerjt) [reply src/common/utils/uaa/client.go]: will do, thanks!

**Pull request merged**